### PR TITLE
[SHELL32][SDK] Implement ShortSizeFormatW

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Implementing missing features...
JIRA issue: N/A

## Proposed changes

- Modify `shell32.spec`.
- Add `ShortSizeFormatW` prototype into `undocshell.h`.
- Implement `ShortSizeFormatW` function.

### CODE REVIEW

**Review by JoachimHenze — APPROVED** [review]
What do you intend to use it for? Wait for 2nd approval please.

**Comment by whindsaks** [comment]
It is one of the undocumented functions from NT4 that is still around today. Is anyone calling it? I don't know.

**Comment by whindsaks** [comment]
> I would hope not, with the unchecked buffer size

All the original shell functions assume MAX_PATH.

File: dll/win32/shell32/utils.cpp
Comment (HBelusca) [inline dll/win32/shell32/utils.cpp:38]: Where does the magical "0x8FFF" come from? I mean, why isn't it 0x9000 or some other size?

File: sdk/include/reactos/undocshell.h
Comment (HBelusca) [inline sdk/include/reactos/undocshell.h:687]: Please use multi-line formatting, like the function prototype that is above.

File: dll/win32/shell32/utils.cpp
Comment (whindsaks) [reply dll/win32/shell32/utils.cpp:38]: > Where does the magical "0x8FFF" come from?

Very strange. The output is always going to be pretty short, "1023 bytes", "5.32 GB" etc. `0x8FFF` makes no sense, it is not a UNICODE_STRING limit or any other kind of limit I have ever heard of.

File: sdk/include/reactos/undocshell.h
Comment (katahiromz) [reply sdk/include/reactos/undocshell.h]: Fixed.

**Comment by katahiromz** [comment]
Should be Okay. Let's go.

File: dll/win32/shell32/utils.cpp
Comment (whindsaks) [reply dll/win32/shell32/utils.cpp]: @katahiromz Why can't you explain where this limit comes from and why it needs to be this large?

Comment (katahiromz) [reply dll/win32/shell32/utils.cpp]: The size unit depends on locale info.
The system locale info can be hacked.
So it is unpredictable.

The original implementation uses this value.

**Pull request merged**
